hawkbit-repository/hawkbit-repository-jpa-hibernate/src/main/java/org/eclipse/hawkbit/repository/jpa/model/AbstractJpaTenantAwareBaseEntity.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hawkbit.repository.jpa.model;
+
+import java.io.Serial;
+import java.util.Objects;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.persistence.PrePersist;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.eclipse.hawkbit.repository.exception.TenantNotExistException;
+import org.eclipse.hawkbit.repository.jpa.model.helper.TenantAwareHolder;
+import org.eclipse.hawkbit.repository.model.TenantAwareBaseEntity;
+import org.hibernate.annotations.TenantId;
+
+/**
+ * Holder of the base attributes common to all tenant aware entities.
+ */
+@NoArgsConstructor(access = AccessLevel.PROTECTED) // Default constructor needed for JPA entities.
+@Setter
+@Getter
+@MappedSuperclass
+public abstract class AbstractJpaTenantAwareBaseEntity extends AbstractJpaBaseEntity implements TenantAwareBaseEntity {
```

Comment:
currently all entities seem to extend eclipse link's AbstractJpaTenantAwareBaseEntity by default, and eclipse-link is the default vendor... still trying to figure out how future modifications/refactorings of these 2 entites ensure compile/compatibility .. maybe separate runs of build with eclipse link and hibernate ?

Reply:
These abstract classes are pretty stable - probably not modified since the beginning.
Having verification actions for both providers will make PR to fail if inconsistency occurs.